### PR TITLE
README header for KIC 2.0 public beta

### PR DESCRIPTION
**Pull request by mflendrich**
Related to https://github.com/Kong/docs.konghq.com/pull/3099#issuecomment-896968376

Adds a section to the top of the readme to announce that KIC 2.0 is in public beta, including a link to the interim documentation site.

**Do not merge** before https://github.com/Kong/kubernetes-ingress-controller/pull/1689 is released and the deploy manifests run the newly released version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +14,21 @@ Custom Resource Definitions(CRDs) and Kubernetes-native tooling.
 
 [**Features**](#features) | [**Get started**](#get-started) | [**Documentation**](#documentation) | [**main branch builds**](#main-branch-builds) | [**Seeking help**](#seeking-help)
 
+## ✨ KIC 2.0 is in public beta! ✨
+
+We're happy to announce that Kong Ingress Controller 2.0 is in public beta.
+
+This new release comes with a major internal rearchitecture, and the following added features:
```
Comment (ccfishk) [inline README.md:21]:
```suggestion
This 2.0 beta release comes with a major internal re-architecture, and the newly added features:
```

File: README.md [hunk 2]
```diff
@@ -16,6 +14,21 @@ Custom Resource Definitions(CRDs) and Kubernetes-native tooling.
...
+- Support for UDP ingress resources and the new `UDPIngress` CRD,
+- Ingress controller runtime metrics compatible with Prometheus, for easier monitoring and alerting.
+
+For a quick start, just deploy KIC 2.0 beta to your cluster:
+```console
```
Comment (ccfishk) [inline README.md:26]:
```suggestion
```deployment
```

**Review by ccfishk — CHANGES_REQUESTED** [review]
looks good

File: README.md (hunk 2 above)
Comment (mflendrich) [reply README.md:26]: Why?

Per GH documentation, there is no such syntax coloring as `deployment`.
https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks
https://github.com/github/linguist/blob/master/lib/linguist/languages.yml

File: README.md (hunk 1 above)
Comment (mflendrich) [reply README.md:21]: I don't understand why any of these changes would be helpful.

On _rearchitecture_ vs _re-architecture_: https://en.wiktionary.org/wiki/rearchitecture

Comment (ccfishk) [reply README.md:21]: isn't 2.0 beta accurate than "new release" ? 
according to english syntax https://english.stackexchange.com/questions/325149/re-architecture, I think you mean to change the design or architecture (of a technical system)

File: README.md (hunk 2 above)
Comment (ccfishk) [reply README.md:26]: Deployment is more accurate, user could execute the tool either on a console or on a terminal.
This is a "README.md", no a code block.

**Review by ccfishk — CHANGES_REQUESTED** [review]
no blocker, wording.

**Review by shaneutt — APPROVED** [review]

**Pull request merged**